apps/ai-image-generator/backend/src/actions/aiig-generate-image.ts

```diff
@@ -6,12 +6,30 @@ interface AppActionCallParameters {
   prompt: string;
 }
 
-interface AppActionResponse {
-  status: number;
-  prompt: string;
-  images: string[];
+interface Image {
+  url: string;
 }
 
+// TODO: Create generic versions of success and error
+export interface AppActionCallResponseSuccess {
+  ok: true;
+  data: {
+    type: 'ImageCreationResult';
+    images: Image[];
+  };
+}
+
+export interface AppActionCallResponseError {
+  ok: false;
+  error: {
+    type: string;
+    message: string;
+    details: Record<string, any>;
```

Comment:
This opens up a lot of interesting options. Would be curious to see what your intention is down the line when we start formalizing and tackling the error side of things in a few weeks.